flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/AbstractStreamOperator.java

```diff
@@ -416,8 +416,10 @@ public final OperatorSnapshotFutures snapshotState(long checkpointId, long times
 			String snapshotFailMessage = "Could not complete snapshot " + checkpointId + " for operator " +
 				getOperatorName() + ".";
 
-			LOG.info(snapshotFailMessage, snapshotException);
-			throw new Exception(snapshotFailMessage, snapshotException);
+			if (!getContainingTask().isCanceled()) {
+				LOG.info(snapshotFailMessage, snapshotException);
+				throw new Exception(snapshotFailMessage, snapshotException);
```

Comment:
I think we should throw the exception in any case, so as to not return any value here. That also matches the previous behaviour, only the logging was added and needs to be guarded, the exception is already silenced.